v6-glm-py/central.py

```diff
@@ -0,0 +1,634 @@
+"""
+This file contains all central algorithm functions. It is important to note
+that the central method is executed on a node, just like any other method.
+
+The results in a return statement are sent to the vantage6 server (after
+encryption if that is enabled).
+"""
+
+from functools import reduce
+import numpy as np
+import pandas as pd
+import scipy.stats as stats
+
+from vantage6.algorithm.tools.util import info, warn, get_env_var
+from vantage6.algorithm.tools.decorators import algorithm_client
+from vantage6.algorithm.client import AlgorithmClient
+from vantage6.algorithm.tools.exceptions import UserInputError, AlgorithmExecutionError
+
+from .common import Family, get_formula
+from .constants import (
+    DEFAULT_MAX_ITERATIONS,
+    DEFAULT_TOLERANCE,
+    ENVVAR_MINIMUM_ORGANIZATIONS,
+    DEFAULT_MINIMUM_ORGANIZATIONS,
+    SIGNIFICANT_DIGITS_FINAL_OUTPUT,
+)
+
+
+@algorithm_client
+def glm(
+    client: AlgorithmClient,
+    family: str,
+    outcome_variable: str | None = None,
+    predictor_variables: list[str] | None = None,
+    formula: str | None = None,
+    categorical_predictors: list[str] = None,
+    category_reference_values: dict[str, str] = None,
+    survival_sensor_column: str = None,
+    tolerance_level: float = DEFAULT_TOLERANCE,
+    max_iterations: int = DEFAULT_MAX_ITERATIONS,
+    organizations_to_include: list[int] = None,
+) -> dict:
+    """
+    Central part of the GLM algorithm
+
+    This function creates subtasks for all the organizations involved in the GLM
+    computation to compute partial results on their data and aggregates these, over
+    multiple cycles, until the GLM is converged.
+
+    Parameters
+    ----------
+    client : AlgorithmClient
+        The client object to interact with the server
+    family : str, optional
```

Comment:
This does not seem to be *optional* looking at the function signature

Reply:
You're right it's not. Updated docstring